spec/javascripts/modules/sticky-element-container.spec.js

```diff
@@ -2,62 +2,72 @@ describe('A sticky-element-container module', function () {
   'use strict'
 
   var GOVUK = window.GOVUK
-  var $ = window.$
-  var instance
 
-  beforeEach(function () {
-    instance = new GOVUK.Modules.StickyElementContainer()
-  })
+  describe('on desktop', function () {
+    var $element
+    var $footer
+    var instance
+
+    beforeEach(function () {
+      $element = $(
+        '<div data-module="sticky-element-container" style="height: 9001px; margin-bottom: 1000px">' +
+          '<div data-sticky-element>' +
+            '<span>Content</span>' +
+          '</div>' +
+        '</div>'
+      )
+
+      instance = new GOVUK.Modules.StickyElementContainer($element[0])
+      $footer = $element.find('[data-sticky-element]')
+
+      instance.getWindowDimensions = function () {
+        return {
+          height: 768,
+          width: 1024
+        }
+      }
+    })
 
-  describe('in a large parent element', function () {
-    var $element = $(
-      '<div data-module="sticky-element-container" style="height: 9001px; margin-bottom: 1000px">' +
-        '<div data-sticky-element>' +
-          '<span>Content</span>' +
-        '</div>' +
-      '</div>'
-    )
-    var $footer = $element.find('[data-sticky-element]')
-
-    describe('on desktop', function () {
-      beforeEach(function () {
-        instance.getWindowDimensions = function () {
-          return {
-            height: 768,
-            width: 1024
-          }
+    it('hides the element, when scrolled at the top', function () {
+      instance.getWindowPositions = function () {
+        return {
+          scrollTop: 0
         }
-      })
+      }
 
-      it('hides the element, when scrolled at the top', function () {
-        instance.start($element)
+      instance.checkResize()
+      instance.checkScroll()
```

Comment:
Why do these two functions now need to be called where they weren't previously? I can see this in a few of the tests - is this related to not calling the `init` function?

Reply:
Yeah, this and your previous comment are related to the `init()` function.  What I found was that the tests do not test the actual scrolling behaviour - in essence they just test some basic maths and the ability to add/remove classes on an element.  I've tested both with and without the `init()` function in the `beforeEach()` and the behaviour is exactly the same.  Which leads me to believe it has no value being run in the tests.

Probably out of scope of this PR - but should we create some tests that actually do test the physical scrolling behaviour?